_posts/2018-05-02-load-balanced-smart-proxies-with-haproxy.md

```diff
@@ -0,0 +1,151 @@
+
+Repeat this step for all smart proxies that are going to sit behind the same load balancer.
+
+### Preparing `custom-hiera.yaml`
+These steps are identical to the steps used on the first Smart Proxy.
+
+### Running the installer
+As only the first Smart Proxy will host a PuppetCA, we point the Puppet agent to it explicitly during the installation and also configure the Puppet certificate to include the `smartproxy.example.net` name:
+```
+[root@smartproxy02] # foreman-installer \
+…
+--puppet-dns-alt-names smartproxy.example.net \
+--puppet-ca-server proxy01.example.net \
+--foreman-proxy-puppetca false \
+--puppet-server-ca false
+```
+## Installing HAProxy
+
+Please note that if you are using a different load balancer technology, you can use this configuration as reference.
+
+Configure SELinux to run in `permissive` mode.
+
+The HAProxy setup itself is super simple, just load-balance a bunch of ports in TCP mode round-robin across all the Smart Proxies. TCP mode is required because the load balancer will not be doing TLS termination. There are just two exceptions to that: port 443 should use `source` not `roundrobin` to not confuse Yum, and the additional port for the PuppetCA (8141) is forwarded only to the first smart proxy.
+
+| Service    | Port | Mode | Balance mode | Destination |
+| ---------- | ---- | ---- | ------------ | ----------- |
+| HTTP       | 80   | TCP  | `roundrobin` | port 80 on all proxies |
+| Anaconda   | 8000 | TCP  | `roundrobin` | port 8000 on all proxies|
+| HTTPS      | 443  | TCP  | `source`     | port 443 on all proxies |
+| RHSM       | 8443 | TCP  | `roundrobin` | port 8443 on all proxies |
+| AMQP       | 5647 | TCP  | `roundrobin` | port 5647 on all proxies |
+| Puppet     | 8140 | TCP  | `roundrobin` | port 8140 on all proxies |
+| PuppetCA   | 8141 | TCP  | `roundrobin` | port 8140 on the first proxy |
+| SmartProxy | 9090 | TCP  | `roundrobin` | port 9090 on all proxies |
+| Docker     | 5000 | TCP  | `roundrobin` | port 5000 on all proxies |
+
+[You can find an example `haproxy.cfg` in the `forklift` repository.](https://github.com/theforeman/forklift/blob/master/roles/haproxy/templates/haproxy.cfg.j2)
+
+## Configuring Clients
+
+As far as clients are concerned, they should only know about `smartproxy.example.net`, and should never need to connect directly to a smartproxy. That has to be configured for Puppet, goferd, and for RHSM/Yum.
+
+### Using katello-client-bootstrap
+
+[katello-client-bootstrap](https://github.com/Katello/katello-client-bootstrap) is designed to configure a machine to be used with an exiting Katello environment. It can also configure Puppet properly for us and supports [setting a separate Puppet CA and Puppet CA port in current Git](https://github.com/Katello/katello-client-bootstrap/pull/250).
+
+To attach clients to the load-balanced proxy, we just have to pass `--server smartproxy.example.net` and `--puppet-ca-port 8141`, `bootstrap.py` will take care of the rest.
```

Comment:
Did you mean `--server` instead of `--puppet-server` here?

Reply:
`--server` is correct, as that will use that name for all things registration (API, katello-ca-consumer RPM, puppet, puppetCA), whereas `--puppet-server` is only to set a puppet server that is *different* from the one passed in `--server`